### PR TITLE
CLN: avoid catching Exception in io.pytables

### PR DESCRIPTION
**Pull request by jbrockmendel**
Last one in this file for a bit, scout's honor.

### CODE REVIEW

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -4242,38 +4243,29 @@ def write_data_chunk(self, rows, indexes, mask, values):
             if not np.prod(v.shape):
                 return
 
-        try:
```
Comment (WillAyd) [inline pandas/io/pytables.py]: So just nothing hitting these in the test cases right? If so, any insight from git blame as to why this was here in the first place?

Comment (jbrockmendel) [reply pandas/io/pytables.py]: Pretty much, yah.  The blame for these goes to back before we were using github AFAICT, so there arent associated PRs/threads.

**Comment by jreback** [comment]
thanks @jbrockmendel 

btw you can blame me for most of this code :->

**Comment by jbrockmendel** [comment]
> btw you can blame me for most of this code :->

One that I'm having trouble with you might be able to help: I'd like to make data_columns, index_axes, non_index_axes, etc into properties, but it isn't clear to me what the underlying logic of how they are defined is.  maybe it depends on version via create_for_block?

**Comment by jreback** [comment]
> > btw you can blame me for most of this code :->
> 
> One that I'm having trouble with you might be able to help: I'd like to make data_columns, index_axes, non_index_axes, etc into properties, but it isn't clear to me what the underlying logic of how they are defined is. maybe it depends on version via create_for_block?

data_columns is defined when writing a table (user passed + index columns from the passed frame); or is index columns on the passed frame if not passed 

the others are computed from this, meaning 

columns in the stored frame + index columns  = index_axes = data_columns + non_index_axes

HTH

**Pull request merged**